akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -111,6 +112,18 @@ object Source {
   }
 
   /**
+   * Creates [[Source]] with `start` as the first element and each next element as `previous + 1` until
+   * it reaches `end`. It allows to create `Source` out of range as simply as on scala `Source(1 to N)`
+   */
+  def range(start: Int, end: Int): javadsl.Source[Integer, Unit] = {
+    require(start <= end, "start must be less or equal than end")
+    from(new util.AbstractList[Integer]() {
+      override def get(index: Int) = start + index
+      override def size() = end - start + 1
```

Comment:
To be symmetric with something like Source.apply(() => Iterator), it should be a toString override.